internal/kibana/slo.go

```diff
@@ -0,0 +1,477 @@
+
+	return resourceSloRead(ctx, d, meta)
+}
+
+func resourceSloUpdate(ctx context.Context, d *schema.ResourceData, meta interface{}) diag.Diagnostics {
+	client, diags := clients.NewApiClient(d, meta)
+	if diags.HasError() {
+		return diags
+	}
+
+	slo, diags := getSloFromResourceData(d)
+	if diags.HasError() {
+		return diags
+	}
+
+	compId, diags := clients.CompositeIdFromStr(d.Id())
+	if diags.HasError() {
+		return diags
+	}
+	slo.ID = compId.ResourceId
+
+	res, diags := kibana.UpdateSlo(ctx, client, slo)
+
+	if diags.HasError() {
+		return diags
+	}
+
+	id := &clients.CompositeId{ClusterId: slo.SpaceID, ResourceId: res.ID}
+	d.SetId(id.String())
+
+	return resourceSloRead(ctx, d, meta)
+}
+
+func resourceSloRead(ctx context.Context, d *schema.ResourceData, meta interface{}) diag.Diagnostics {
+	client, diags := clients.NewApiClient(d, meta)
+	if diags.HasError() {
+		return diags
+	}
+	compId, diags := clients.CompositeIdFromStr(d.Id())
+	if diags.HasError() {
+		return diags
+	}
+	id := compId.ResourceId
+	spaceId := compId.ClusterId
+
+	s, diags := kibana.GetSlo(ctx, client, id, spaceId)
+	if s == nil && diags == nil {
+		d.SetId("")
+		return diags
+	}
+	if diags.HasError() {
+		return diags
+	}
+
+	indicators := []interface{}{}
+	if s.Indicator.IndicatorPropertiesApmAvailability != nil {
+		params := s.Indicator.IndicatorPropertiesApmAvailability.Params
+		indicators = append(indicators, map[string]interface{}{
+			"type": s.Indicator.IndicatorPropertiesApmAvailability.Type,
+			"params": []map[string]interface{}{{
+				"environment":      params.Environment,
+				"service":          params.Service,
+				"transaction_type": params.TransactionType,
+				"transaction_name": params.TransactionName,
+				"index":            params.Index,
+				"filter":           params.Filter,
+			}},
+		})
+	} else if s.Indicator.IndicatorPropertiesApmLatency != nil {
+		params := s.Indicator.IndicatorPropertiesApmLatency.Params
+		indicators = append(indicators, map[string]interface{}{
+			"type": s.Indicator.IndicatorPropertiesApmLatency.Type,
+			"params": []map[string]interface{}{{
+				"environment":      params.Environment,
+				"service":          params.Service,
+				"transaction_type": params.TransactionType,
+				"transaction_name": params.TransactionName,
+				"index":            params.Index,
+				"filter":           params.Filter,
+				"threshold":        params.Threshold,
+			}},
+		})
+	} else if s.Indicator.IndicatorPropertiesCustomKql != nil {
+		params := s.Indicator.IndicatorPropertiesCustomKql.Params
+		indicators = append(indicators, map[string]interface{}{
+			"type": s.Indicator.IndicatorPropertiesCustomKql.Type,
+			"params": []map[string]interface{}{{
+				"index":           params.Index,
+				"filter":          params.Filter,
+				"good":            params.Filter,
+				"total":           params.Total,
+				"timestamp_field": params.TimestampField,
+			}},
+		})
+	} else {
+		return diag.Errorf("unknown indicator type")
```

Comment:
Na, since I'm just looking to see which indicator type is set. If it hits here, the case is actually that the indicator is somehow unset. Will update log message.